scripts/tests.unit.sh

```diff
@@ -9,13 +9,12 @@ if ! [[ "$0" =~ scripts/tests.unit.sh ]]; then
   exit 255
 fi
 
-HYPERSDK_PATH=$(
-  cd "$(dirname "${BASH_SOURCE[0]}")"
-  cd .. && pwd
-)
-# shellcheck source=/scripts/constants.sh
-source "$HYPERSDK_PATH"/scripts/constants.sh
+SCRIPT_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+source "${SCRIPT_DIR}"/constants.sh
 
-# Provision of the list of tests requires word splitting, so disable the shellcheck
-# shellcheck disable=SC2046
-go test -race -timeout="10m" -coverprofile="coverage.out" -covermode="atomic" $(find . -name "*.go" | grep -v "./x/programs/cmd" | grep -v "./examples/morpheusvm" | xargs -n1 dirname | sort -u | xargs)
+file_args=()
+while IFS= read -r line; do
+    file_args+=("$line")
+done < <(find . -type f -name "*.go" | grep -v "./x/programs/cmd" | grep -v "./examples/morpheusvm" | xargs -n1 dirname | sort -u)
+
+go test -race -timeout="10m" -coverprofile="coverage.out" -covermode="atomic" "${file_args[@]}"
```

Comment:
Avoid word split, for more information see `Exceptions` at https://www.shellcheck.net/wiki/SC2046
`readarray` would have been more readable but it's unsupported on macos because it uses an outdated version of bash for licensing reasons.